packages/ComponentLibrary/src/components/Breadcrums/index.tsx

```diff
@@ -0,0 +1,220 @@
+                      handleMiddleMenuClose();
+                    }}
+                    sx={sx.menuItem}>
+                    {item.label}
+                  </MenuItem>
+                ))}
+              </Menu>
+            </Box>
+          )}
+          {renderBreadcrumbItem(lastItem, true)}
+        </>
+      );
+    }
+  };
+  const renderBreadcrumbItem = (item: BreadcrumbItem, isLast: boolean) => (
+    <Box key={item.id} sx={sx.breadcrumbItem}>
+      {isLast ? (
+        <>
+          <Typography
+            noWrap
+            sx={sx.lastItemTypography}
+            onClick={() => {
+              if (item.actions && item.actions.length > 0) {
+                handleActionMenuOpen(item.actions);
+              }
+            }}>
+            {item.label}
+          </Typography>
+          {item.actions && item.actions.length > 0 && (
+            <IconButton
+              size="small"
+              ref={actionButtonRef}
+              onClick={() => handleActionMenuOpen(item.actions!)}
+              sx={sx.actionButton}>
+              <ChevronDown fill={theme.palette.baselineColor.neutral[80]} />
+            </IconButton>
+          )}
+        </>
+      ) : (
+        <Link
+          href="#"
+          onClick={e => {
+            e.preventDefault();
+            item.onClick && item.onClick();
+          }}
+          sx={sx.link}>
+          <Typography noWrap sx={sx.breadcrumbTypography}>
+            {item.label}
+          </Typography>
+        </Link>
+      )}
+    </Box>
+  );
+
+  return (
+    <Box sx={sx.container}>
+      <Breadcrumbs
+        separator={separator}
+        aria-label="breadcrumb"
+        sx={sx.breadcrumbs}>
+        <Box sx={sx.homeContainer}>
+          <Link
+            href="#"
+            onClick={e => {
+              e.preventDefault();
+              onHomeClick();
+            }}
+            onMouseEnter={() => setIsHomeHovered(true)}
+            onMouseLeave={() => setIsHomeHovered(false)}
+            sx={sx.homeLink}>
+            {renderHomeIcon()}
+          </Link>
+          <Typography onClick={onHomeClick} sx={sx.homeText}>
+            {homeText}
+          </Typography>
+        </Box>
+        {renderBreadcrumbItems()}
+      </Breadcrumbs>
+      <Menu
+        anchorEl={anchorEl}
+        open={Boolean(anchorEl)}
+        onClose={handleActionMenuClose}
+        slotProps={{
+          paper: { sx: sx.menu, elevation: 3 },
+        }}
+        MenuListProps={{ sx: menuStyle }}>
+        {currentActions.map(action => (
+          <MenuItem key={action.id} onClick={() => {}} sx={sx.menuItem}>
+            <Box sx={sx.iconBox}>
+              {action.icon}
+              <span>{action.label}</span>
+            </Box>
+            {action.toggle && (
+              <Box sx={sx.toggleContainer}>
+                <ToggleChip
+                  isActive={toggleStates[action.id] ?? false}
+                  onToggle={() => handleToggle(action.id)}
```

Comment:
note to future ourselves: we need to develop some helper to create this handlers inside map loops and avoid the usage of arrow functions